app/main/forms/suppliers.py

```diff
@@ -50,7 +50,9 @@ class DunsNumberForm(Form):
 class CompaniesHouseNumberForm(Form):
     companies_house_number = StripWhitespaceStringField('Companies house number', validators=[
         Optional(),
-        Length(min=8, max=8, message="Companies House numbers must have 8 characters.")
+        Regexp(r'^([0-9]{2}|[A-Za-z]{2})[0-9]{6}$',
+               message="Companies House numbers must have 8 digits or 2 letters followed by 6 digits."
```

Comment:
Could this message be confusing? `12345678123456` and `AB123456` both fit the description.

Reply:
Hmm, perhaps. Words are hard.

Reply:
"Companies House numbers must start with either 8 digits or 2 letters, followed by 6 digits."

Or something.

Reply:
I did have an "either" in there before but deleted it thinking it was superfluous.  What do you reckon to "Companies House numbers must have *either* 8 digits or 2 letters followed by 6 digits." - better?

Reply:
Again, could be `12345678` or `AB123456`. I think the comma separates the two parts out.

To be honest, it's probably fine though. I'm just nit picking. Whatever you think is best.

Reply:
Wait, ignore me. I'm very tired. Yours is good.